substrate/client/network/transactions_v2/transactions_v2_proposal.md

```diff
@@ -0,0 +1,166 @@
+- The descriptor begins with a format specifier, allowing various types of payload:
+  - `0`: transaction body v1: `leb128(size tx body) ++ scale-encoded tx-body` (LEB128-encoded size followed by SCALE-encoded transaction body, enabling direct transmission of the full transaction body),
+  - `1`: 32-byte hash v1 (currently used),
+  - `2`: 32-bit transaction fingerprint v1, allowing further bandwidth optimizations, and laying the ground for *PinSketch* based set-reconciliation implementation as defined in *Erlay*.
+
+Two latter payloads are referred as transaction identifiers.
+
+#### Transactions Identifiers Flooding (`TxIF`).
+Transaction identifiers are gossiped to all connected peers (except _LightNodes_) allowing many transaction descriptors in a single networking notification. This reduces the required bandwidth and the depth of queues in networking module implementation. Identifiers are transmitted after a transaction is submitted to the local pool and validated as ready.
+
+The transaction descriptor shall not be sent to a peer that already knows its identifier, whether because we received it from that peer or previously sent the transaction descriptor to it. Known identifiers shall be kept for T seconds, with a maximum of N identifiers per peer. Networks should configure these limits based on their transaction volumes, and node network and memory requirements.
+
+Once a transaction descriptor is received, and it is not a transaction body, the latter shall be downloaded using the `TxRR` protocol (see the next section) and imported to the local pool. If a received transaction descriptor contains the transaction data it shall be imported to the pool.
+
+If the transaction associated with recieved desriptor is found to be invalid, the reputation of peer gossiping invalid desciptor shall be decreased.
+
+_Notes_:
+- some measures shall be taken to avoid sending single identifier in notification. The [pool import notification stream](https://github.com/paritytech/polkadot-sdk/blob/ec700de9cdca84cdf5d9f501e66164454c2e3b7d/substrate/client/service/src/builder.rs#L593) shall be drained (maybe with some reasonable delay - 30-50ms to speculatively allow more transactions to come).
+- when the combined size of all identifiers to be gossiped is relatively small compared to the network packet size, the network notification can include some of the transaction bodies to reduce the latency required for transaction dissemination. It is up to implementation to decide what is the most efficient approach.
+
+#### Transaction Data Request-Response Protocol (`TxRR`).
+After receiving a transaction identifier, the node should request the transaction body from a random peer which gossiped the transaction identifier, if the transaction is not already in the local pool, and there is not a pending request for it. Requests should have a short timeout to avoid denial of service where peers only gossip identifiers (and never provide their bodies).
```

Comment:
nit: We should specify the timeout, however it's difficult to accurately measure the timeout since the protocol accepts batching as well

Reply:
Maybe const component + component depending on requested transactions count?
So we can give the other peer some more time when the request is bulkier?

added this in: af2ee969d8